src/Model/Context.php

```diff
@@ -98,11 +117,11 @@ public function has($name)
     }
 
     /**
-     * @return array
+     * @return array|\stdClass
      */
     public function jsonSerialize()
     {
-        return $this->data;
+        return $this->data ?: new \stdClass;
```

Comment:
Ok, the thing I wanted to make sure is that the Context does not get serialized into an array, since that would be serialized the wrong way.
In your PR you have a check in place where we do not send an empty Context. So for that matter it would be fine to just return the data array with no check at all. Alright with that? I then would restore that line to the previous state.